### PR TITLE
Slack: consolidate offer-accepted message, comma off thousands

### PR DESCRIPTION
**Pull request by duncanjbrown**
## Context

We get up to three messages per offer acceptance in Slack, as the candidate might automatically decline and withdraw at the same time.

Also, numbers in the thousands are difficult to read without commas, and we'll soon hit 10,000 candidates (!).

## Changes proposed in this pull request

- Move the `without_slack_message_sending` method onto `StateChangeNotifier` where it belongs and give it a better name
- Consolidate the offer-accepted message
- Comma off thousands

I think `offer accepted` and `application sent to provider` are the only state changes entailing multiple messages, so I didn't make a generic `buffer_slack_messages` method. (That might be been usable in both cases but the messages would have been less readable without some fiddly extra work of the sort already in this PR but more unpleasantly generic).

## Guidance to review

Tests should prove it works 🤞

## Things to check

- [X] This code does not rely on migrations in the same Pull Request
- [X] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [X] API release notes have been updated if necessary
- [X] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/lib/state_change_notifier.rb [hunk 1]
```diff
@@ -23,6 +26,33 @@ def self.submit_application(application_form)
...
+        "#{ac.offered_option.course.name_and_code} at #{ac.offered_option.course.provider.name}"
+      end
+
+      withdrawn_msg += withdrawals.to_sentence
+    end
+
+    if declined.any?
+      declined_msg = 'declined '
+      declines = declined.map do |ac|
+        "#{ac.offered_option.course.provider.name}’s offer on #{ac.offered_option.course.name_and_code}"
```
Comment (benswannack) [inline app/lib/state_change_notifier.rb:44]: Is `offer on x` the right wording here? I'd expect it to read `offer for x`. (Same thing for accepted message above)

**Review by benswannack — APPROVED** [review]
Nice 👍

File: app/lib/state_change_notifier.rb (hunk 1 above)
Comment (duncanjbrown) [reply app/lib/state_change_notifier.rb:44]: Good point — looks like we use `for` elsewhere. I'll update :+1:

Comment (duncanjbrown) [reply app/lib/state_change_notifier.rb]: Updated

**Pull request merged**